lib/storage/drivers/s3.ts

```diff
@@ -17,24 +17,36 @@ import { streamToBuffer } from '~/lib/utils'
 export const s3Driver = defineStorageDriver({
   envSchema: z.object({
     STORAGE_S3_BUCKET: z.string().min(1),
-    STORAGE_S3_ENDPOINT: z.string().min(1),
+    STORAGE_S3_ENDPOINT: z.string().optional(),
     STORAGE_S3_REGION: z.string().min(1).default('us-east-1'),
-    STORAGE_S3_PORT: z.coerce.number().positive(),
-    STORAGE_S3_USE_SSL: z.string().transform((v) => v === 'true'),
-    STORAGE_S3_ACCESS_KEY: z.string().min(1),
-    STORAGE_S3_SECRET_KEY: z.string().min(1),
+    STORAGE_S3_PORT: z.coerce.number().positive().optional(),
+    STORAGE_S3_USE_SSL: z
+      .string()
+      .transform((v) => v === 'true')
+      .optional(),
+    STORAGE_S3_ACCESS_KEY: z.string().optional(),
+    STORAGE_S3_SECRET_KEY: z.string().optional(),
+    AWS_REGION: z.string().optional(),
+    AWS_DEFAULT_REGION: z.string().optional(),
```

Comment:
My suggestion to please both sides would be to use Zods discriminated union around a new parameter (how about `STORAGE_S3_AUTO_CREDENTIALS`?).
When this parameter is set to 'true', then all the other parameters could be optional, otherwise it would function normal for the manual approach.

Reply:
https://zod.dev/?id=discriminated-unions